fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/mixin/registry/sync/SimpleRegistryMixin.java

```diff
@@ -382,4 +401,104 @@ public void unmap(String name) throws RemapException {
 			fabric_prevEntries = null;
 		}
 	}
+
+	@Override
+	public void addAlias(Identifier old, Identifier newId) {
+		Objects.requireNonNull(old, "alias cannot be null");
+		Objects.requireNonNull(newId, "aliased id cannot be null");
+
+		if (fabric_aliases.containsKey(old)) {
+			throw new IllegalArgumentException(
+					"Tried adding %s as an alias for %s, but it is already an alias (for %s) in registry %s".formatted(
+							old,
+							newId,
+							fabric_aliases.get(old),
+							this.key
+					)
+			);
+		} else if (this.idToEntry.containsKey(old)) {
+			throw new IllegalArgumentException(
+					"Tried adding %s as an alias, but it is already present in registry %s".formatted(
+							old,
+							this.key
+					)
+			);
+		} else if (getAliased(newId).equals(old)) {
+			// since an alias corresponds to at most one identifier, this is the only way to create a cycle
+			// that doesn't already fall under the first condition
+			throw new IllegalArgumentException(
+					"Making %1$s an alias of %2$s would create a cycle, as %2$s is already an alias of %1$s (registry %3$s)".formatted(
+							old,
+							newId,
+							this.key
+					)
+			);
+		} else if (!this.idToEntry.containsKey(newId)) {
+			FABRIC_LOGGER.warn(
+					"Adding {} as an alias for {}, but the latter doesn't exist in registry {}",
+					old,
+					newId,
+					this.key
+			);
+		}
+
+		fabric_aliases.put(old, newId);
+	}
+
+	@Unique
+	private Identifier getAliased(Identifier id) {
+		Identifier alias = id;
+
+		while (fabric_aliases.containsKey(alias)) {
+			alias = fabric_aliases.get(alias);
+		}
+
+		return alias;
+	}
+
+	@Unique
+	private RegistryKey<T> getAliased(RegistryKey<T> key) {
+		Identifier aliased = getAliased(key.getValue());
+		return aliased == key.getValue() ? key : RegistryKey.of(key.getRegistryRef(), aliased);
+	}
+
+	@WrapMethod(method = "get(Lnet/minecraft/registry/RegistryKey;)Ljava/lang/Object;")
```

Comment:
It would probably be better to use `@ModifyVariable(at = @At("HEAD"), argsOnly = true)` to modify the parameter rather than wrapping the method.

Reply:
Sorry didn't get this ping for whatever reason, but I agree with Earth. It would also allow all the ones which have the same parameter type to be merged into 1 handler.